Helper/Order.php

```diff
@@ -886,17 +898,21 @@ public function updateOrderPayment($order, $transaction = null, $reference = nul
             case self::TS_AUTHORIZED:
                 $orderState = OrderModel::STATE_PROCESSING;
                 $transactionType = Transaction::TYPE_AUTH;
-                $transactionId = $transaction->id.'-auth';
+                if ($prevTransactionState == self::TS_AUTHORIZED) {
+                    $transactionId = $transaction->id.'-capture-'.time();
+                    $parentTransactionId = $transaction->id.'-auth';
+                }else{
+                    $transactionId = $transaction->id.'-auth';
+                }
 
                 break;
             case self::TS_COMPLETED:
                 $orderState = OrderModel::STATE_PROCESSING;
-                $transactionId = $transaction->id.'-capture';
                 $amount = $transaction->capture->amount->amount;
 
                 if ($prevTransactionState == self::TS_AUTHORIZED) {
                     $transactionType = Transaction::TYPE_CAPTURE;
-                    $transactionId = $transaction->id.'-capture';
+                    $transactionId = $transaction->id.'-capture-'.time();
                     $parentTransactionId = $transaction->id.'-auth';
                 } else {
                     $transactionType = Transaction::TYPE_PAYMENT;
```

Comment:
Isn't setting of $transactionId ($transaction->id.'-capture') needed here?
Assuming it is not necessary, can stay at default,  $transaction->id, but maybe more precise name shown in admin can have its benefits.

Reply:
I didn't touch this part, other than making it unique since there will now be more than one. Is there a reason you think it should only be "$transaction->id.'-capture'"?

Reply:
@johnmccombs1 what I meant is putting $transactionId = $transaction->id.'-capture'; in `else {` section

Reply:
Currently it's set to:
$transactionId = $transaction->id.'-payment';

Does it need to change to:
$transactionId = $transaction->id.'-capture';

Just let me know and I'll update it.